### PR TITLE
Support long press callback for the Link component

### PR DESCRIPTION
**Pull request by hccoelho**
This change allow to pass a longpress callback with a url as parameter (e.g. it can be used for menu options of the url)

Please review @erictraut @berickson1 @deregtd

### CODE REVIEW

File: src/web/Link.tsx [hunk 1]
```diff
@@ -84,10 +92,28 @@ export class Link extends RX.Link<void> {
...
+                if (this.props.onLongPress) {
+                    this.props.onLongPress(e, this.props.url);
+                }
+            }, _longPressTime);
+        }
+    }    
+
+    private _onMouseUp = (e: Types.SyntheticEvent) => {
+        if (this._longPressTimer) {
+            window.clearTimeout(this._longPressTimer);
```
Comment (erictraut) [inline src/web/Link.tsx]: It's good form to set this._longPressTimer to undefined after clearing it.

File: src/web/Link.tsx [hunk 2]
```diff
@@ -37,7 +37,14 @@ const _styles = {
     }
 };
 
+const _longPressTime = 1000;
+
 export class Link extends RX.Link<void> {
+    
+    private _lastMouseDownEvent: Types.SyntheticEvent;
```
Comment (erictraut) [inline src/web/Link.tsx]: This doesn't appear to be used.

File: src/web/Link.tsx [hunk 3]
```diff
@@ -37,7 +37,14 @@ const _styles = {
     }
 };
 
+const _longPressTime = 1000;
+
 export class Link extends RX.Link<void> {
+    
+    private _lastMouseDownEvent: Types.SyntheticEvent;    
+    private _longPressTimer: number;    
+    private _ignoreClick = false;
```
Comment (erictraut) [inline src/web/Link.tsx]: Same here.

File: src/web/Link.tsx (hunk 1 above)
Comment (hccoelho) [inline src/web/Link.tsx]: Good call.

File: src/web/Link.tsx (hunk 2 above)
Comment (hccoelho) [inline src/web/Link.tsx]: Grrrr. Some left overs from another thing I wanted to include but not right now.

**Comment by hccoelho** [comment]
@erictraut Thanks for the good suggestions and for catching some leftovers. Updated the PR. Let me know if there is anything else.

**Pull request merged**